kitsune/users/middleware.py

```diff
@@ -17,3 +19,22 @@ def process_request(self, request):
 
             logout(request)
             return HttpResponseRedirect(reverse('home'))
+
+
+class LogoutInvalidatedSessionsMiddleware(MiddlewareMixin):
+    """Logs out any sessions started before a user changed their
+    Firefox Accounts password.
+    """
+    def process_request(self, request):
+
+        user = request.user
+
+        if user.is_authenticated():
```

Comment:
Let's add a TODO here for python3 which will be merged right after this. We already have an issue to track the TODOs